phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
Ok, single method proposal.

One single method, called `not_enableable`.  All current uses of `is_enableable` should simply be negated, and that would be 100% same behaviour as now.

If the method returns `true` the extension cannot be enabled, without any specified reason, so the default error string should be used.  If the method returns a (non-empty) string, or a (non-empty) array of strings, then this should be interpreted as PROCESSED language strings, and transmitted as such to the end user.  If the method returns false, an empty string or an empty array, the extension may be enabled normally.

The `base` implementation would implement `not_enableable` as `!is_enableable`, therefore making it 100% BC.  For extensions that have implemented a "workaround" for this limitation, they will need to rewrite, or continue using the "workaround".

To help extension developers with the language processing of the returned string/array, a helper function will be included in the `base` ext object, so that the extension developer may simply do `return $this->helper($array_of_key)`, being the array of keys as complex as described above, with all possible variations.

If this proposal is accepted, I will gladly code it and resubmit, but I won't do anything until the concept is approved.  If this proposal is not accepted, please, let me know, so that I can close this PR.